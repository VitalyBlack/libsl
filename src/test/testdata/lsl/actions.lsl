libsl "1.0.0";
library simple;

types {
   Int(int);
}

automaton A : Int {
   var i: Int;

   fun f(param: Int) {
      action TEST_ACTION(1, "123", param, 1+123);
      action TEST_ACTION_TWO();
   }
}