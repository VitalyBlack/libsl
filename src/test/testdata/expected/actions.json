{
  "name": "simple",
  "lslVersion": "1.0.0",
  "types": [
    {
      "name": "Int",
      "realName": "int"
    }
  ],
  "automata": [
    {
      "name": "A",
      "constructorVariables": [],
      "variables": [
        {
          "name": "i",
          "type": "Int"
        }
      ],
      "states": [],
      "shifts": [],
      "functions": [
        {
          "name": "f",
          "automaton": "A",
          "args": [],
          "contracts": [],
          "statements": [
            {
              "kind": "action",
              "actionName": "TEST_ACTION",
              "args": [
                {
                  "kind": "integer",
                  "value": 1
                },
                {
                  "kind": "string",
                  "value": "123"
                },
                {
                  "kind": "variableAccess",
                  "variable": "f.param"
                }
              ]
            },
            {
              "kind": "action",
              "actionName": "TEST_ACTION_TWO",
              "args": []
            }
          ]
        }
      ]
    }
  ]
}